aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandlerTest.java

```diff
@@ -43,83 +53,226 @@
+            assertThat(exception).hasCauseExactlyInstanceOf(ServiceQuotaExceededException.class);
+        }
+
+        @Test
+        public void testInternalServerException() {
+            doThrow(InternalServerException.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(any(), any());
+
+            CfnServiceInternalErrorException exception = assertThrows(CfnServiceInternalErrorException.class, () ->
+                    subject.handleRequest(proxy, request, null, logger));
+            assertThat(exception).hasCauseExactlyInstanceOf(InternalServerException.class);
+        }
+
+        @Test
+        public void testThrottlingException() {
+            doThrow(ThrottlingException.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(any(), any());
 
-        assertThrows(CfnThrottlingException.class, () ->
-                new CreateHandler().handleRequest(proxy, request, null, logger));
+            CfnThrottlingException exception = assertThrows(CfnThrottlingException.class, () ->
+                    subject.handleRequest(proxy, request, null, logger));
+            assertThat(exception).hasCauseExactlyInstanceOf(ThrottlingException.class);
+        }
+
+        @Test
+        public void testValidationException() {
+            doThrow(ValidationException.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(any(), any());
+
+            CfnInvalidRequestException exception = assertThrows(CfnInvalidRequestException.class, () ->
+                    subject.handleRequest(proxy, request, null, logger));
+            assertThat(exception).hasCauseExactlyInstanceOf(ValidationException.class);
+        }
+
+        @Test
+        public void testAnyOtherCodeGuruExceptionException() {
+            doThrow(CodeGuruProfilerException.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(any(), any());
+
+            CodeGuruProfilerException exception = assertThrows(CodeGuruProfilerException.class, () ->
+                    subject.handleRequest(proxy, request, null, logger));
+            assertThat(exception).hasNoCause();
+        }
     }
 
-    @Test
-    public void testValidationException() {
-        doThrow(ValidationException.builder().build())
-                .when(proxy).injectCredentialsAndInvokeV2(any(), any());
+    private ResourceModel newResourceModel(final AgentPermissions permissions) {
+        return ResourceModel.builder()
+                .profilingGroupName(profilingGroupName)
+                .agentPermissions(permissions)
+                .build();
+    }
 
-        assertThrows(CfnInvalidRequestException.class, () ->
-                new CreateHandler().handleRequest(proxy, makeInvalidRequest(), null, logger));
+    private void assertResponse(ProgressEvent<ResourceModel, CallbackContext> response) {
+        assertThat(response).isNotNull();
```

Comment:
Minor: This can be removed -- other tests will also validate this one implicitly.

Reply:
Maybe, but if the response is indeed _null_, it would fail faster and clearer, in my opinion.